lib/plug/cowboy/conn.ex

```diff
@@ -90,6 +90,11 @@ defmodule Plug.Cowboy.Conn do
     :cowboy_req.inform(status, to_headers_map(headers), req)
   end
 
+  @impl true
+  def upgrade(req, :websocket, opts), do: {:ok, req |> Map.put(:upgrade, {:websocket, opts})}
```

Comment:
My goal was to call this function `adapter_upgrade`, exactly because the underlying semantics are all dictated by the adapter.